bindings/rust/s2n-tls-tokio/src/config.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use s2n_tls::raw::{config, connection::Connection, enums::Mode, error::Error};
+
+/// Produces new, un-negotiated connections.
+pub trait Config {
+    fn create(&self, mode: Mode) -> Result<Connection, Error>;
```

Comment:
Yeah `Config` does seem a little weird here, I think as you suggested `connection::Provider` might be more appropriate, and the method could be `fn new_connection()` just to make it extra clear

Reply:
Renaming to "Builder", but not connection::Builder. Also after some more thinking / discussion, I'm going to rework this to handle TlsStreams instead of Connections. However, that will happen in a separate PR, since this fix does unblock other development work.